subsys/net/lib/http/http_server_ws.c

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <strings.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/net/http/service.h>
+#include <zephyr/sys/base64.h>
+#include <mbedtls/sha1.h>
+
+LOG_MODULE_DECLARE(net_http_server, CONFIG_NET_HTTP_SERVER_LOG_LEVEL);
+
+#include "headers/server_internal.h"
+
+#if !defined(ZEPHYR_USER_AGENT)
+#define ZEPHYR_USER_AGENT "Zephyr"
+#endif
+
+/* From RFC 6455 chapter 4.2.2 */
+#define WS_MAGIC "258EAFA5-E914-47DA-95CA-C5AB0DC85B11"
+
+/* Handle upgrade from HTTP/1.1 to Websocket, see RFC 6455
+ */
+int handle_http1_to_websocket_upgrade(struct http_client_ctx *client)
+{
+	static const char switching_protocols[] =
+		"HTTP/1.1 101 Switching Protocols\r\n"
+		"Connection: Upgrade\r\n"
+		"Upgrade: websocket\r\n"
+		"Sec-WebSocket-Accept: ";
+	char key_accept[32 + sizeof(WS_MAGIC)];
```

Comment:
Is this 32 related to `HTTP_SERVER_WS_MAX_SEC_KEY_LEN`?

Reply:
Yes, that is correct. I will change this to use the macro.